packages/next/src/lib/metadata/resolve-title.ts

```diff
@@ -0,0 +1,25 @@
+import type { Metadata } from './types/metadata-interface'
+import type { AbsoluteTemplateString } from './types/metadata-types'
+
+function resolveTitleTemplate(template: string, title: string) {
+  return template.replace(/%s/g, title)
+}
+
+export function resolveTitle(
+  stashed: AbsoluteTemplateString,
+  title: Metadata['title']
+) {
+  if (typeof title === 'string') {
+    stashed.absolute = resolveTitleTemplate(stashed.template, title)
+  } else if (title) {
+    if ('default' in title) {
+      stashed.absolute = resolveTitleTemplate(stashed.template, title.default)
+    }
+    if ('absolute' in title) {
+      stashed.absolute = resolveTitleTemplate(stashed.template, title.absolute)
```

Comment:
if you are providing an absolute title via an `AbsoluteTemplateString` we should not be templatizing it